linera-core/src/updater.rs

```diff
@@ -200,47 +202,24 @@ where
     A: ValidatorNode + Clone + 'static,
     S: Storage + Clone + Send + Sync + 'static,
 {
-    async fn send_optimized_certificate(
-        &mut self,
-        certificate: &Certificate,
-        delivery: CrossChainMessageDelivery,
-    ) -> Result<Box<ChainInfo>, NodeError> {
-        if certificate.is_signed_by(&self.remote_node.name) {
-            let result = self
-                .remote_node
-                .handle_lite_certificate(certificate.lite_certificate(), delivery)
-                .await;
-            match result {
-                Err(NodeError::MissingCertificateValue) => {
-                    warn!(
-                        "Validator {} forgot a certificate value that they signed before",
-                        self.remote_node.name
-                    );
-                }
-                _ => {
-                    return result;
-                }
-            }
-        }
-        self.remote_node
-            .handle_certificate(certificate.clone(), vec![], delivery)
-            .await
-    }
-
     async fn send_certificate(
         &mut self,
         certificate: Certificate,
         delivery: CrossChainMessageDelivery,
     ) -> Result<Box<ChainInfo>, NodeError> {
         let result = self
-            .send_optimized_certificate(&certificate, delivery)
+            .remote_node
+            .handle_optimized_certificate(&certificate, delivery)
```

Comment:
OK, I thought the `ValidatorUpdater` is a part of code that updates the local view about the chain using a remote node. Why do we ask a `remote_node` to handle a certificate then? Shouldn't we use a local node?

Reply:
No, the `ValidatorUpdater` is mainly about updating a _remote_ node, i.e. sending stuff to the validators.